model/post_list_test.go

```diff
@@ -69,6 +69,34 @@ func TestPostListExtend(t *testing.T) {
 	} else if len(l2.Posts) != 3 || len(l2.Order) != 3 {
 		t.Fatal("extending l2 again changed l2")
 	}
+
+	// Extend where length of Posts and Order is not equal
+	// It can happen when a PostList contains parent post/s
+	l3 := PostList{}
+
+	l3.AddPost(p1)
+	l3.AddOrder(p1.Id)
+
+	l3.AddPost(p2)
+	l3.AddOrder(p2.Id)
+
+	l4 := PostList{}
+
+	l4.AddPost(p3)
+	l4.AddOrder(p3.Id)
+
+	p4 := &Post{Id: NewId(), Message: NewId()}
+	l4.AddPost(p4)
+
+	l4.Extend(&l3)
+
+	if len(l3.Posts) != 2 || len(l3.Order) != 2 {
```

Comment:
Would you mind using https://github.com/stretchr/testify for this?

Reply:
Thanks @hanzei and sure.  Test is now updated.